lib/src/mistralai_client_dart_base.dart

```diff
@@ -19,8 +24,38 @@ class MistralAIClient {
   final int maxRetries;
   final http.Client _client;
 
+  Future<String> _getRequest<T>(String endpoint) async {
+    final headers = <String, String>{
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer $apiKey',
+    };
+
+    try {
+      final response = await _client
+          .get(Uri.parse('$baseUrl$endpoint'), headers: headers)
+          .timeout(timeout);
+
+      if (response.statusCode != 200) {
+        throw Exception('Get request failed: ${response.statusCode}');
+      }
+
+      return response.body;
+    } catch (e) {
+      throw Exception('Get request failed: $e');
+    }
+  }
```

Comment:
I would move most of this into `listModels`. `HTTP GET` is only used in the `listModels` API call. That's why I don't think we need a separate `get` wrapper method.

The only parts that will be the same are error handling and headers but we will clean it up after the initial implementation of public methods I think.

Wdyt?

Reply:
Now I see that the `embedding` method will use post request. I will move this logic to `listModels` method and maybe we will do something with it after implementing all public methods in this class.  Is that ok for you?